New package version: mfem v4.4

v-dobrev opened the pull request:
Modifications:
- [x] Test CUDA builds
- [x] Test HIP builds
- [x] Test with all external dependencies

v-dobrev committed:
New package version: mfem v4.4

v-dobrev committed:
MFEM 4.4: update versions of dependencies; add HYPRE GPU flags

eugeneswalker commented:
I think this needs a rebase to overcome known broken specs -- this looks to be why the data-viz-sdk CI generation step failed.

v-dobrev committed:
MFEM 4.4: more tweaks in the package

v-dobrev committed:
MFEM 4.4: handle the 'camp' dependency of 'raja'

v-dobrev committed:
Merge branch 'develop' into packages/mfem-4.4

v-dobrev commented:
> I think this needs a rebase to overcome known broken specs -- this looks to be why the data-viz-sdk CI generation step failed.

I merged `develop` into this branch -- that should have the same effect as rebasing.

v-dobrev committed:
SUNDIALS: add the BLAS/LAPACK dependencies of the hypre dependency

v-dobrev committed:
PETSc: when using older versions, do not try to install files
       that did not exist in the older version.

v-dobrev committed:
MUMPS: add support for building with gfortran 10+ when usnig mixed
       toolchains like Apple Clang C/C++ with gfortran.

v-dobrev committed:
MFEM: more tweaks in the dependencies and the testing script.

v-dobrev committed:
Merge branch 'develop' into packages/mfem-4.4

Resolved conflict:
   var/spack/repos/builtin/packages/sundials/package.py

alalazo commented:
@v-dobrev Just fyi, since a while you can convert the PR to [draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests#draft-pull-requests) instead of marking it WIP

v-dobrev committed:
MFEM: propagate 'cuda_arch' to hypre, petsc, slepc; more tweaks to
      the testing script.

v-dobrev committed:
Merge branch 'develop' into packages/mfem-4.4

v-dobrev committed:
MFEM: fix spack style

v-dobrev committed:
MFEM: update test script

v-dobrev committed:
STRUMPACK: small fix to NOT add 'slate' dependency with '~slate'.

v-dobrev committed:
PETSC: when 'cflags', 'fflags', and/or 'cxxflags' are not set, let
       the PETSc configuration choose these flags.

v-dobrev committed:
HYPRE: when building with '+rocm', unset 'CFLAGS' and 'CXXFLAGS'
       which are currently set by 'llvm-amdgpu', so that they do
       not override hypre's default flags.

v-dobrev committed:
MFEM: some additions and tweaks, mostly related to HIP support.

v-dobrev committed:
Merge branch 'develop' into packages/mfem-4.4

v-dobrev commented:
This PR is now ready for review. (I removed the `WIP` label.)

Edit: Oops, typo: "is not" --> "is now". One letter completely inverted the meaning. 😄

tzanio reviewed (dismissed):
Thank you @v-dobrev

v-dobrev committed:
MFEM: adjust testing script: cuda tests with libceed now work but
      take too long.

v-dobrev commented:
@tzanio, @acfisher, can you please re-approve -- my last commit (that changes just the tesing script `.../mfem/test_builds.sh`) marked your reviews as stale.

jeremylt commented on var/spack/repos/builtin/packages/mfem/package.py:
```diff
@@ -257,29 +297,34 @@ class Mfem(Package, CudaPackage, ROCmPackage):
+    depends_on('libceed@0.6', when='@:4.1+libceed')
+    depends_on('libceed@0.7:0.8', when='@4.2.0+libceed')
+    depends_on('libceed@0.8:0.9', when='@4.3.0+libceed')
+    depends_on('libceed@0.10:', when='@4.4.0:+libceed')
```
```suggestion
    depends_on('libceed@0.10.1:', when='@4.4.0:+libceed')
```

v-dobrev committed:
MFEM: update libCEED version dependency

Co-authored-by: Jeremy L Thompson <jeremy@jeremylt.org>

v-dobrev committed:
Merge branch 'develop' into packages/mfem-4.4

v-dobrev committed:
MFEM: update the conflict with '^mpich@4:' to mfem <= 4.3

v-dobrev commented:
@tzanio, @acfisher, as we discussed offline, I updated the conflict with `^mpich@4:` to mfem <= 4.3.

If all the CI checks pass, can you please re-approve one more time?

tzanio commented:
Thanks @v-dobrev !
